### PR TITLE
OTLP LogExporter modified to not drop whole batch if a single log is invalid

### PR DESCRIPTION
**Pull request by cijothomas**
Would be nice follow ups to log Exception detail as opposed to just message in this PR.

### CODE REVIEW

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/Implementation/ExportClient/OtlpHttpTraceExportClientTests.cs [hunk 1]
```diff
@@ -32,7 +32,7 @@
 using Xunit;
 using OtlpCollector = Opentelemetry.Proto.Collector.Trace.V1;
 
-namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests.Implementation.ExportClient
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests
```
Comment (cijothomas) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/Implementation/ExportClient/OtlpHttpTraceExportClientTests.cs:35]: unrelated cleanup

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,12 @@
   are now configurable via the `MetricReaderOptions`.
   ([#2717](https://github.com/open-telemetry/opentelemetry-dotnet/pull/2717))
 
+* Exporter bug fix to not throw exceptions from Export method.
+  ([#2915](https://github.com/open-telemetry/opentelemetry-dotnet/pull/2915))
+
+* OTLP LogExporter modified to not drop whole batch if a single log from the
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md:20]:
```suggestion
* OTLP LogExporter modified to not drop the whole batch if a single log from the
```

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs [hunk 3]
```diff
@@ -46,77 +47,81 @@ internal static class LogRecordExtensions
             foreach (var item in logRecordBatch)
             {
                 var logRecord = item.ToOtlpLog();
-                if (logRecord == null)
+                if (logRecord != null)
```
[the remark this answers is not in the document]
Comment (reyang) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs:50]: I see, I was confused by the name, consider renaming it to `otlpLogRecord`.

**Review by reyang — APPROVED** [review]
LGTM assuming the changelog will be updated to include the PR link.
